standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/MetaStoreAuditLogBuilder.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hive.metastore;
+
+import java.util.List;
+import java.util.Map;
+
+import org.apache.hadoop.hive.common.TableName;
+
+import static org.apache.commons.lang3.StringUtils.join;
+
+/**
+ * Generate the audit log in a builder manner.
+ */
+public class MetaStoreAuditLogBuilder {
+  // the function
+  private final String functionName;
+  private final StringBuilder builder;
+
+  private MetaStoreAuditLogBuilder(String functionName) {
+    this.functionName = functionName;
+    this.builder = new StringBuilder();
+  }
+
+  public static MetaStoreAuditLogBuilder functionName(String functionName) {
+    MetaStoreAuditLogBuilder builder = new MetaStoreAuditLogBuilder(functionName);
+    return builder;
+  }
+
+  public MetaStoreAuditLogBuilder connectorName(String connectorName) {
+    builder.append("connector=").append(connectorName).append(" ");
+    return this;
+  }
+
+  public MetaStoreAuditLogBuilder catalogName(String catalogName) {
+    builder.append("catName=").append(catalogName).append(" ");
+    return this;
+  }
+
+  public MetaStoreAuditLogBuilder dbName(String dbName) {
+    builder.append("db=").append(dbName).append(" ");
+    return this;
+  }
+
+  public MetaStoreAuditLogBuilder tableName(String tableName) {
+    builder.append("tbl=").append(tableName).append(" ");
+    return this;
+  }
+
+  public MetaStoreAuditLogBuilder packageName(String packageName) {
+    builder.append("package=").append(packageName).append(" ");
+    return this;
+  }
+
+  public MetaStoreAuditLogBuilder typeName(String typeName) {
+    builder.append("type=").append(typeName).append(" ");
+    return this;
+  }
+
+  public MetaStoreAuditLogBuilder pattern(String pattern) {
+    builder.append("pat=").append(pattern).append(" ");
+    return this;
+  }
+
+  public MetaStoreAuditLogBuilder extraInfo(String extraInfo) {
```

Comment:
all the callsites of this method look like:
```
" some=" + value
```
they not only expose the internal "key=value" strucutre; but they should also add and extra whitespace.

I think it would be better to add arguments like `key, value` - and let the builder take up some responsibility.

one more thing: `extraInfo` is pasted directly into the output builder; meanwhile for other methods the `key=` is written by this class.

Reply:
Make sense, let me take a look. Thank you!